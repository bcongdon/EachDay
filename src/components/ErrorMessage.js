import React, { Component } from 'react'
import { Message } from 'semantic-ui-react'
import { PropTypes } from 'prop-types'

class ErrorMessage extends Component {
  static propTypes = {
    message: PropTypes.string.isRequired,
    header: PropTypes.string,
    compact: PropTypes.bool
  }

  render() {
    const header = this.props.header === undefined ? 'Something went wrong...' : this.props.header
    return (
      <Message negative compact={this.props.compact}>
        <Message.Header>{header}</Message.Header>
        <p>{this.props.message}</p>
      </Message>
    )
  }
}

export default ErrorMessage
