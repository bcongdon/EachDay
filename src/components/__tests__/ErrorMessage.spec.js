import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ErrorMessage from '../ErrorMessage'

it('renders correctly', () => {
  const message = shallow(
    <ErrorMessage message='foo' />
  )
  expect(toJson(message)).toMatchSnapshot()
})

it('renders correctly with custom header', () => {
  const message = shallow(
    <ErrorMessage message='foo' header='Something blew up!' />
  )
  expect(toJson(message)).toMatchSnapshot()
})

it('renders correctly with compact', () => {
  const message = shallow(
    <ErrorMessage message='foo' compact />
  )
  expect(toJson(message)).toMatchSnapshot()
})
