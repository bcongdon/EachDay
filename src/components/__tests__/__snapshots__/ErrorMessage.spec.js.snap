// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Message
  negative={true}
>
  <MessageHeader>
    Something went wrong...
  </MessageHeader>
  <p>
    foo
  </p>
</Message>
`;

exports[`renders correctly with compact 1`] = `
<Message
  compact={true}
  negative={true}
>
  <MessageHeader>
    Something went wrong...
  </MessageHeader>
  <p>
    foo
  </p>
</Message>
`;

exports[`renders correctly with custom header 1`] = `
<Message
  negative={true}
>
  <MessageHeader>
    Something blew up!
  </MessageHeader>
  <p>
    foo
  </p>
</Message>
`;
